openage/util/fslike/path.py

```diff
@@ -74,6 +77,20 @@ def __repr__(self):
             return repr(self.fsobj) + ".root"
 
         return f"Path({repr(self.fsobj)}, {repr(self.parts)})"
+    
+    @staticmethod
+    def create_temp_file():
+        # Create a temporary file using tempfile.NamedTemporaryFile
+        temp_file = tempfile.NamedTemporaryFile(delete=False)
+        # Wrap the temporary file path in a Path object and return it
+        return Path(temp_file.name)
+
+    @staticmethod
+    def create_temp_dir():
+        # Create a temporary directory using tempfile.mkdtemp
+        temp_dir = tempfile.mkdtemp()
+        # Wrap the temporary directory path in a Path object and return it
+        return Path(temp_dir)
```

Comment:
you need to set `is_temp` to `True` in this method

Reply:
Changed

Reply:
no, please don't do this via `__init__`. You should set it to true in the `create_temp_file` and `create_temp_dir` functions.

Reply:
like should I pass them into the functions or just add them inside the function?

Reply:
you don't need to pass anything ^^

you need to create the path, then set the member, then return it.

```
path = Path(..)
path.is_temp = True
return path
```

Reply:
oh.... :'(
Changed

Reply:
Hmm... still not right. Maybe my comments are a bit confusing :D

You should return the `path` that you created. Right now, you are still returning a different path.